k8s/gvclient.go

```diff
@@ -32,26 +34,50 @@ type groupVersionClient struct {
 
 func (g *groupVersionClient) get(ctx context.Context, identifier resource.Identifier, plural string,
 	into resource.Object) error {
+	ctx, span := GetTracer().Start(ctx, "kubernetes-get")
+	defer span.End()
 	sc := 0
 	request := g.client.Get().Resource(plural).Name(identifier.Name)
 	if strings.TrimSpace(identifier.Namespace) != "" {
 		request = request.Namespace(identifier.Namespace)
 	}
 	bytes, err := request.Do(ctx).StatusCode(&sc).Raw()
+	span.SetAttributes(
+		attribute.Int("http.response.status_code", sc),
+		attribute.String("http.request.method", http.MethodGet),
+		attribute.String("server.address", request.URL().Hostname()),
+		attribute.String("server.port", request.URL().Port()),
+		attribute.String("url.full", request.URL().String()),
+	)
 	if err != nil {
-		return parseKubernetesError(bytes, sc, err)
+		err = parseKubernetesError(bytes, sc, err)
+		span.SetStatus(codes.Error, err.Error())
+		return err
 	}
-	return rawToObject(bytes, into)
+	err = rawToObject(bytes, into)
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+	}
+	return err
 }
 
 func (g *groupVersionClient) getMetadata(ctx context.Context, identifier resource.Identifier, plural string) (
 	*k8sObject, error) {
+	ctx, span := GetTracer().Start(ctx, "kubernetes-getmetadata")
+	defer span.End()
 	sc := 0
 	request := g.client.Get().Resource(plural).Name(identifier.Name)
 	if strings.TrimSpace(identifier.Namespace) != "" {
 		request = request.Namespace(identifier.Namespace)
 	}
 	bytes, err := request.Do(ctx).StatusCode(&sc).Raw()
+	span.SetAttributes(
+		attribute.Int("http.response.status_code", sc),
+		attribute.String("http.request.method", http.MethodGet),
+		attribute.String("server.address", request.URL().Hostname()),
+		attribute.String("server.port", request.URL().Port()),
+		attribute.String("url.full", request.URL().String()),
+	)
 	if err != nil {
 		return nil, parseKubernetesError(bytes, sc, err)
```

Comment:
Do we want to set the span status to `error` here? Similar to how it's done for `*groupVersionClient.get()` here https://github.com/grafana/grafana-app-sdk/blob/7caf8e0298fcb2b760914c3b5b9fe76304b479f9/k8s/gvclient.go#L54